src/Avalonia.Controls/SplitButton/SplitButton.cs

```diff
@@ -0,0 +1,471 @@
+
+                // If flyout is changed while one is already open, make sure we 
+                // close the old one first
+                // This is the same behavior as Button
+                if (oldFlyout != null &&
+                    oldFlyout.IsOpen)
+                {
+                    oldFlyout.Hide();
+                }
+
+                // Must unregister events here while a reference to the old flyout still exists
+                if (oldFlyout != null)
+                {
+                    UnregisterFlyoutEvents(oldFlyout);
+                }
+
+                if (newFlyout != null)
+                {
+                    RegisterFlyoutEvents(newFlyout);
+                }
+
+                UpdatePseudoClasses();
+            }
+
+            base.OnPropertyChanged(e);
+        }
+
+        /// <inheritdoc/>
+        protected override void OnKeyDown(KeyEventArgs e)
+        {
+            var key = e.Key;
+
+            if (key == Key.Space || key == Key.Enter) // Key.GamepadA is not currently supported
+            {
+                UpdatePseudoClasses();
+            }
+
+            base.OnKeyDown(e);
+        }
+
+        /// <inheritdoc/>
+        protected override void OnKeyUp(KeyEventArgs e)
+        {
+            var key = e.Key;
+
+            if (key == Key.Space || key == Key.Enter) // Key.GamepadA is not currently supported
+            {
+                UpdatePseudoClasses();
+
+                // Consider this a click on the primary button
+                if (IsEffectivelyEnabled)
+                {
+                    OnClickPrimary(null);
+                    e.Handled = true;
+                }
+            }
+            else if (key == Key.Down && IsEffectivelyEnabled)
+            {
+                // WinUI requires the VirtualKey.Menu (alt) + VirtualKey.Down to open the flyout
+                // Avalonia will only require Key.Down which is better cross-platform
+
+                OpenFlyout();
+                e.Handled = true;
+            }
+            else if (key == Key.F4 && IsEffectivelyEnabled)
+            {
+                OpenFlyout();
+                e.Handled = true;
+            }
+            else if (e.Key == Key.Escape && _isFlyoutOpen)
+            {
+                // If Flyout doesn't have focusable content, close the flyout here
+                // This is the same behavior as Button
+                CloseFlyout();
+                e.Handled = true;
+            }
+
+            base.OnKeyUp(e);
+        }
+
+        /// <summary>
+        /// Invokes the <see cref="Click"/> event when the primary button part is clicked.
+        /// </summary>
+        /// <param name="e">The event args from the internal Click event.</param>
+        protected virtual void OnClickPrimary(RoutedEventArgs e)
+        {
+            // Note: It is not currently required to check enabled status; however, this is a failsafe
+            if (IsEffectivelyEnabled)
+            {
```

Comment:
Original event args should be handled at this point. So it won't be bubbled to the parents.

With reused Button.ClickEvent it also will be raised twice, if original event wasn't handled.
OnClickSecondary also should do the same.

Reply:
See comments above. I don't think re-using is a good idea.